libsupport/src/EntityTypeManager.cpp

```diff
@@ -476,54 +476,24 @@ katana::EntityTypeManager::PrintEntityTypes() const {
 bool
 katana::EntityTypeManager::Equals(
     const katana::EntityTypeManager& other) const {
-  if (entity_type_id_to_atomic_entity_type_ids_ !=
-      other.entity_type_id_to_atomic_entity_type_ids_) {
-    KATANA_LOG_DEBUG(
-        "this.entity_type_id_to_atomic_entity_type_ids_.size() = {}, "
-        "other.size() = {}. SetOfEntityTypeIDsSize = {}, "
-        "other.SetOfEntityTypeIDsSize = {}",
-        entity_type_id_to_atomic_entity_type_ids_.size(),
-        other.entity_type_id_to_atomic_entity_type_ids_.size(),
-        SetOfEntityTypeIDsSize_, other.SetOfEntityTypeIDsSize_);
-
-    KATANA_LOG_DEBUG(
-        "this.entity_type_id_to_atomic_entity_type_ids_.at(0).size = {}, "
-        "other.size = "
-        "{}",
-        entity_type_id_to_atomic_entity_type_ids_.at(0).size(),
-        other.entity_type_id_to_atomic_entity_type_ids_.at(0).size());
-    return false;
-  }
-  if (atomic_entity_type_id_to_type_name_ !=
-      other.atomic_entity_type_id_to_type_name_) {
-    KATANA_LOG_DEBUG(
-        "this.atomic_entity_type_id_to_type_name_.size() = {}, other.size() = "
-        "{}",
-        atomic_entity_type_id_to_type_name_.size(),
-        other.atomic_entity_type_id_to_type_name_.size());
-    return false;
-  }
-
-  if (atomic_type_name_to_entity_type_id_ !=
-      other.atomic_type_name_to_entity_type_id_) {
+  auto end = GetNumEntityTypes();
+  auto other_end = other.GetNumEntityTypes();
+  if (end != other_end) {
     KATANA_LOG_DEBUG(
-        "this.atomic_type_name_to_entity_type_id_.size() = {}, other.size() = "
-        "{}",
-        atomic_type_name_to_entity_type_id_.size(),
-        other.atomic_type_name_to_entity_type_id_.size());
+        "this.GetNumEntityTypes() = {}, "
+        "other.GetNumEntityTypes() == {}",
+        end, other_end);
     return false;
   }
-
-  if (atomic_entity_type_id_to_entity_type_ids_ !=
-      other.atomic_entity_type_id_to_entity_type_ids_) {
-    KATANA_LOG_DEBUG(
-        "this.atomic_entity_type_id_to_entity_type_ids_.size() = {}, "
-        "other.size() = {}. SetOfEntityTypeIDsSize = {}, "
-        "other.SetOfEntityTypeIDsSize = {}",
-        atomic_entity_type_id_to_entity_type_ids_.size(),
-        other.atomic_entity_type_id_to_entity_type_ids_.size(),
-        SetOfEntityTypeIDsSize_, other.SetOfEntityTypeIDsSize_);
-    return false;
+  for (size_t i = 0; i < end; ++i) {
```

Comment:
On the surface this doesn't look quite right. 

Shouldn't we also have to ensure that all names in `this.atomic_type_name_to_entity_type_id_.keys()` are present in `other.atomic_type_name_to_entity_type_id_.keys()`? 

From the name of `GetNonAtomicEntityTypeFromStrings` it really looks like we are only checking the managers NonAtomicEntityTypes here. 

Looking closer at `GetNonAtomicEntityTypeFromStrings` When `i` is an AtomicEntityTypeID, aka `tns` only has one name in it. 

`GetEntityTypeIDs` will function properly and give us the identity `SetOfEntityTypeIDs` for the AtomicEntityType name. 
which we pass to `GetNonAtomicEntityType`

the important part being
```
  auto existing_id = std::find(
      entity_type_id_to_atomic_entity_type_ids_.cbegin(),
      entity_type_id_to_atomic_entity_type_ids_.cend(), type_id_set_resized);
```

Since in `entity_type_id_to_atomic_entity_type_ids_` an AtomicEntityTypeID maps to its identity `SetOfEntityTypeIDs` this works. 


I think it would be helpful to add a comment here that we rely on AtomicEntityTypeIds mapping to its "NonAtomic" identity set so it is clear that all EntityTypes (Atomic & NonAtomic) are getting compared?

Reply:
> I think it would be helpful to add a comment here that we rely on AtomicEntityTypeIds mapping to its "NonAtomic" identity set so it is clear that all EntityTypes (Atomic & NonAtomic) are getting compared?

The code in GraphUpdate that uses EntityTypes also relies on this property.  I suspect the partitioner relies on it as well.  I added something along these lines to `GetNonAtomicEntityTypeFromStrings`, but I don't understand the Atomic/NonAtomic design well enough to go any further.